cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/NotificationSubscriberService.java

```diff
@@ -273,11 +272,12 @@ protected void updateJobQueue(DatasetContext context, Notification notification)
         // TODO: (CDAP-11469) No need to check appSpec once migration from DatasetBasedTimeScheduleStore is done
         ApplicationSpecification appSpec = store.getApplication(scheduleId.getParent());
         if (appSpec == null) {
-          throw new ApplicationNotFoundException(scheduleId.getParent());
+          LOG.debug("Cannot find application '{}' for schedule '{}' in AppMetadataStore",
```

Comment:
Is this an indication of a serious inconsistency? or is this expected to happen? Under which circumstances? If iyt is serious, it should log a warning.